### PR TITLE
Lacework EKS-A 

### PR DESCRIPTION
**Pull request by jefferyfry**
*Issue #, if available:*

*Description of changes:*
Added Laceworks HelmRelease and supporting resources for EKS-A validation. New resources are located in:
./eks-anywhere-commons/Testers/Lacework/Polygraph
./eks-anywhere-commons/Addons/Partner/Lacework/Polygraph

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by shapirov103 — COMMENTED** [review]
@jefferyfry thank you for your PR for validation of Lacework components with EKS-A. 

A couple of questions:

1. Have you shared the secrets required to run the tests with AWS?
2. What is the significance of the `flux-system` directory that has a git source pointing to your repo?  Is it a left-over? We don't allow other git repositories to be defined as part of the PRs.

**Comment by jefferyfry** [comment]
Removed the extraneous `flux-system` directory. 

What's the best way for me to share the secrets?

**Comment by shapirov103** [comment]
> What's the best way for me to share the secrets?

Are you on AWS Developers slack? If yes, please share directly with me on DM. Otherwise, please send an email to 
shapirov@amazon.com
elamaras@amazon.com

We can also onboard you onto the AWS Developers Slack for rapid feedback.

File: eks-anywhere-common/Addons/Partner/Lacework/Polygraph/cluster-secret-store.yaml [hunk 1]
```diff
@@ -0,0 +1,19 @@
+apiVersion: external-secrets.io/v1beta1
```
Comment (elamaran11) [inline eks-anywhere-common/Addons/Partner/Lacework/Polygraph/cluster-secret-store.yaml:1]: Please remove this file. Our framework creates `ClusterSecretStore` this to sync secrets in AWS Secrets Manager to the cluster so this is not required.

File: eks-anywhere-common/Testers/Lacework/Polygraph/testJob.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: lacework-agent-testjob
```
Comment (elamaran11) [inline eks-anywhere-common/Testers/Lacework/Polygraph/testJob.yaml:4]: This test job does not qualify for a functional test job. All this does is if pods are coming up. What we need is a functional test job which functionally validates if your software is working fine. Please check out partner submissions on their products for functional test job sample. Functional test job is a criteria for us to proceed with validation.

**Review by elamaran11 — CHANGES_REQUESTED** [review]
@jefferyfry Thankyou so much for the PR. Couple of things to be fixed before we move to validation phase in our labs 
1. `ClusterSecretStore` should be removed
2. Functional test job needs to test the ISV product functionality.

**Review by elamaran11 — CHANGES_REQUESTED** [review]
@jefferyfry Test job is failing. Please see the logs :

```bash
❯ kga -n lacework                                                                                                   ─╯
NAME                               READY   STATUS    RESTARTS   AGE
pod/lacework-agent-6zdpj           1/1     Running   0          4m44s
pod/lacework-agent-cfv7k           1/1     Running   0          4m44s
pod/lacework-agent-g2tth           1/1     Running   0          4m44s
pod/lacework-agent-s6sd8           1/1     Running   0          4m44s
pod/lacework-agent-testjob-6qmfz   0/1     Error     0          3m5s
pod/lacework-agent-testjob-kjh24   0/1     Error     0          3m39s

NAME                            DESIRED   CURRENT   READY   UP-TO-DATE   AVAILABLE   NODE SELECTOR   AGE
daemonset.apps/lacework-agent   4         4         4       4            4           <none>          4m44s

NAME                               COMPLETIONS   DURATION   AGE
job.batch/lacework-agent-testjob   0/1           3m39s      3m39s

░▒▓    /home/eksadmin ▓▒░
❯ k logs lacework-agent-testjob-6qmfz -n lacework                                                                   ─╯
pod/lacework-agent-6zdpj condition met
pod/lacework-agent-cfv7k condition met
pod/lacework-agent-g2tth condition met
pod/lacework-agent-s6sd8 condition met

░▒▓    /home/eksadmin ▓▒░
❯ k logs lacework-agent-testjob-kjh24 -n lacework                                                                   ─╯
pod/lacework-agent-6zdpj condition met
pod/lacework-agent-cfv7k condition met
pod/lacework-agent-g2tth condition met
pod/lacework-agent-s6sd8 condition met
```

**Comment by elamaran11** [comment]
@jefferyfry This does not work with new change too. Im getting same error. I would recommend updating the job to work fine and also testing it in your lab before reaching back. Please share us your job logs of successful run with your next submission.

**Comment by shapirov103** [comment]
Same test job failure on Rover as well. Output stating the condition is met (three times - based on the number of agent pods maybe) then fails with an error status.

**Comment by elamaran11** [comment]
Hi @jefferyfry I ran it now with your latest changes and i can see the test job is working fine in EKS-A Baremetal and i can see the `stats.go` in logs. So i can conclude the test job is working, i will have to run the PR with other deployment models and provide you a full cycle validation completion sometime soon.

```bash
time="2023-08-03T18:28:28.877Z" level=info msg="Pkt (lo)-Stats:{CTMap:290,DNSOut:4,DNSIn:4,DNS-MultiQs:0,Decode-DNS-Err:0,DirOwners:5,DirSyn:88,Drop-Dns-ErrResp:0,DropMal:34,Empty-DNS:0,Map:5,Pcap:1145,PcapSynAck:0,PcapSyn:0,PcapAck:0,PcapFin:0,PcapRst:0,PcapUnk:0,PktRx:2286,Sink-Conn-TCP:93,Sink-Conn-TCP-U:5,Sink-Conn-UDP:0,Sink-Conn-UDP-U:0,Sink-DNS:0,Track:88,ConnCount:93,ConnCountOwned:93,PCAPTickerProTime:{Name:lo PCAP ticker time in ms,UpperBounds:[1,2,4,8,16,32,64,128,256,512,1024,2048,4096,8192,16384,32768,65536,131072,262144,524288,1048576],Counts:[0,0,0,0,0,0,0,0,0,0,1,0,0,0,0,0,0,0,0,0,0,0],CumSum:1004,TotalCount:1,AvgVal:0}}" caller="stats.go:165" pid=914898
time="2023-08-03T18:28:28.878Z" level=info msg="Pkt (cilium_geneve)-Stats:{Bridged:14,CNIDrop:14386,CTMap:290,DNSOut:48,DNSIn:48,DNS-MultiQs:0,Decode-DNS-Err:0,Drop-Dns-ErrResp:0,Empty-DNS:0,Pcap:14502,PcapSynAck:0,PcapSyn:0,PcapAck:0,PcapFin:0,PcapRst:0,PcapUnk:0,PktRx:14045,Sink-Conn-TCP:0,Sink-Conn-TCP-U:0,Sink-Conn-UDP:0,Sink-Conn-UDP-U:0,Sink-DNS:0,PCAPTickerProTime:{Name:cilium_geneve PCAP ticker time in ms,UpperBounds:[1,2,4,8,16,32,64,128,256,512,1024,2048,4096,8192,16384,32768,65536,131072,262144,524288,1048576],Counts:[1,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0],CumSum:0,TotalCount:1,AvgVal:0}}" caller="stats.go:165" pid=914898

 ^‿^ Stats collection is running ^‿^
```

**Review by elamaran11 — APPROVED** [review]
Works on all 4 environments.

**Pull request merged**